docs/connect.html

```diff
@@ -508,7 +508,24 @@ <h4><a id="connect_administration" href="#connect_administration">Kafka Connect
     </p>
 
     <p>
-    When a connector is first submitted to the cluster, the workers rebalance the full set of connectors in the cluster and their tasks so that each worker has approximately the same amount of work. This same rebalancing procedure is also used when connectors increase or decrease the number of tasks they require, or when a connector's configuration is changed. You can use the REST API to view the current status of a connector and its tasks, including the id of the worker to which each was assigned. For example, querying the status of a file source (using <code>GET /connectors/file-source/status</code>) might produce output like the following:
+    When a connector is first submitted to the cluster, a rebalance is triggered between the Connect workers in order to distribute the load that consists of the tasks of the new connector.
+    This same rebalancing procedure is also used when connectors increase or decrease the number of tasks they require, when a connector's configuration is changed, or when a
+    worker is added or removed from the group as part of an intentional upgrade of the Connect cluster or due to a failure.
+    In versions prior to 2.3.0, the Connect workers would rebalance the full set of connectors and their tasks in the cluster as a simple way to make sure that each worker has approximately the same amount of work.
+    This behavior can be still enabled by setting <code>connect.protocol=eager</code>.
+    Starting with 2.3.0, Kafka Connect is using by default a protocol that performs
+    <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-415%3A+Incremental+Cooperative+Rebalancing+in+Kafka+Connect">incremental cooperative rebalancing</a>
+    when connectors and tasks need to be distributed among the Connect workers. Balanced load among workers is still preserved under incremental cooperative rebalancing.
+    But in this case, a task is stopped temporarily during a rebalance only when it needs to move between workers. This transition might be immediate after consecutive lightweight rebalances between workers when a new worker joins.
+    In case a worker leaves the group, intentionally or due to a failure, rebalancing of its tasks is triggered only after a delay equal to the value of the property <code>scheduled.rebalance.max.delay.ms</code>.
+    This delay, that has a default value of five minutes (<code>300000ms</code>), aims to provide a way to tolerate failures or upgrades of workers without immediately redistributing the load of a departing worker.
+    If this worker returns within the configured delay it gets its previous assigned tasks in full. For as long as <code>scheduled.rebalance.max.delay.ms</code> is in effect, these tasks remain unassigned.
+    If a worker does not return in time, the tasks get distributed among the remaining workers in the Connect cluster.
```

Comment:
Some suggestions:

> ... <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-415%3A+Incremental+Cooperative+Rebalancing+in+Kafka+Connect">incremental cooperative rebalancing</a>
that incrementally balances the connectors and tasks across the Connect workers, affecting only tasks that are new, to be removed, or need to move from one worker to another. Other tasks are not stopped and restarted during the rebalance, as they would have been with the old protocol.
>
> If a Connect worker leaves the group, intentionally or due to a failure, Connect waits for <code>scheduled.rebalance.max.delay.ms</code> before triggering a rebalance. This delay defaults to five minutes (<code>300000ms</code>) to tolerate failures or upgrades of workers without immediately redistributing the load of a departing worker. If this worker returns within the configured delay, it gets its previous assigned tasks in full. However, it does mean that the tasks may remain unassigned until the time specified by <code>scheduled.rebalance.max.delay.ms</code> elapses. If a worker does not return within that time limit, Connect will reassign those tasks among the remaining workers in the Connect cluster.

Reply:
Added almost unchanged. Let's re-read.